src/test/java/com/aws/greengrass/clientdevices/auth/iot/ThingTest.java

```diff
@@ -38,6 +41,11 @@ void GIVEN_validThingName_WHEN_Thing_THEN_objectIsCreated() {
         Assertions.assertDoesNotThrow(() -> Thing.of("0123456789"));
     }
 
+    @AfterEach
```

Comment:
Similar thing here. Configure this up front so you have a known state of the world prior to your test running. In this case I think it's probably okay, but it's a dangerous pattern to follow.

Also, nit, move this method before the first test case?

Reply:
addressed